kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,689 @@
+
+            // Compute initial radius (mRadius_0)
+            mpBuilderAndSolver->ResizeAndInitializeVectors(mpScheme, mpA, mpDx, mpb, r_model_part);
+            TSystemMatrixType& mA = *mpA;
+            TSystemVectorType& mDx = *mpDx;
+            TSystemVectorType& mb = *mpb;
+            TSparseSpace::SetToZero(mA);
+            TSparseSpace::SetToZero(mDx);
+            TSparseSpace::SetToZero(mb);
+
+            mpBuilderAndSolver->BuildAndSolve(mpScheme, r_model_part, mA, mDx, mb);
+
+            mRadius_0 = TSparseSpace::TwoNorm(mDx);
+            mRadius = mRadius_0;
+
+            // Compute vector of reference external force (mf)
+            this->InitializeSystemVector(mpf);
+            TSystemVectorType& mf = *mpf;
+            TSparseSpace::SetToZero(mf);
+
+            // We build it now to only include external loads
+            mpBuilderAndSolver->BuildRHS(mpScheme, r_model_part, mf);
+
+            //Initialize the loading factor Lambda
+            mLambda = 0.0;
+            mLambda_old = 1.0;
+
+            // Initialize Norm of solution
+            mNormxEquilibrium = 0.0;
+
+            mInitializeArcLengthWasPerformed = true;
+
+            KRATOS_INFO("Arc-Length Strategy") << "Strategy Initialized" << std::endl;
+        }
+
+		if (!BaseType::mSolutionStepIsInitialized) {
+            BaseType::InitializeSolutionStep();
+            SaveInitializeSystemVector(mpf);
+            InitializeSystemVector(mpDxf);
+            InitializeSystemVector(mpDxb);
+            InitializeSystemVector(mpDxPred);
+            InitializeSystemVector(mpDxStep);
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief Performs all the required operations that should be done (for each step) after solving the solution step.
+     * @details A member variable should be used as a flag to make sure this function is called only once per step.
+     */
+    void FinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        DofsArrayType& r_dof_set = mpBuilderAndSolver->GetDofSet();
+        ModelPart& r_model_part  = BaseType::GetModelPart();
+        TSystemMatrixType& r_A   = *mpA;
+        TSystemVectorType& r_Dx  = *mpDx;
+        TSystemVectorType& r_b   = *mpb;
+
+        unsigned int iteration_number = r_model_part.GetProcessInfo()[NL_ITERATION_NUMBER];
+        // Update the radius
+        mRadius = mRadius*std::sqrt(double(mDesiredIterations)/double(iteration_number));
+        if (mRadius > mMaxRadiusFactor*mRadius_0)
+            mRadius = mMaxRadiusFactor*mRadius_0;
+        else if (mRadius < mMinRadiusFactor*mRadius_0)
+            mRadius = mMinRadiusFactor*mRadius_0;
+
+        mpScheme->FinalizeSolutionStep(r_model_part, r_A, r_Dx, r_b);
+        mpBuilderAndSolver->FinalizeSolutionStep(r_model_part, r_A, r_Dx, r_b);
+        mpConvergenceCriteria->FinalizeSolutionStep(r_model_part, r_dof_set, r_A, r_Dx, r_b);
+        //Cleaning memory after the solution
+        mpScheme->Clean();
```

Comment:
again, why reimplementing and not calling the baselcass?

Reply:
solved